tests/cases/fourslash/server/autoImportProvider_importsMap1.ts

```diff
@@ -0,0 +1,32 @@
+/// <reference path="../fourslash.ts"/>
+
+// @Filename: /tsconfig.json
+//// {
+////   "compilerOptions": {
+////     "module": "nodenext",
+////     "rootDir": "src",
+////     "outDir": "dist"
+////   }
+//// }
+
+// @Filename: /package.json
+//// {
+////   "type": "module",
+////   "imports": {
+////     "#is-browser": {
+////       "browser": "./dist/env/browser.js",
```

Comment:
Does the implementation handle declaration files too? E.g.

```
"browser": {
  "types": "./dist/env/browser.d.ts"
  "default": "./dist/env/browser.js"
}
```

Reply:
This one works OK but I don't think it's thanks to any special logic for declaration files. It likely just reaches the `default` here somehow and makes the mapping for it.

What exact guarantees you'd like to see here when it comes to declaration files?

Reply:
Hah, if we’re reaching `default`, that’s an instance of #50762—I wasn’t sure whether that bug is present in module specifier generation or just module resolution. The output file mapping should work for declaration files too. You could duplicate this test and just change the `.js` extensions to `.d.ts`—I only included the `types`/`default` pair in my example to make it a bit more realistic.

It should also `compilerOptions.declarationDir` as an output location. Looking at the the module resolution code, it doesn’t seem to map declaration files only from `declarationDir` and JS files only from `outDir`—it looks like both directories are considered as a place where an output file could be, and all file extensions are tried in all output directories 🤷

Reply:
> extensionSwappedTarget && comparePaths(extensionSwappedTarget, pathOrPattern) === Comparison.EqualTo

Yep, I confirmed that it's an issue here as well.

I pushed out what I believe was the requested change - could you take a look if that does match your expectations?

Reply:
This actually broke a few tests and I'm investigating that 😅